### DISCUSSION TITLE
How to get mailparse and other PHP extensions working?

### DISCUSSION THREAD

**Discussion by tomderoo** [ddev/ddev#4683] [Q&A] [answered]
Hello all

I need to collaborate on a web project that has the following required php extensions (excerpt from composer.json):

`"ext-bcmath": "*",
"ext-curl": "*",
"ext-dom": "*",
"ext-fileinfo": "*",
"ext-intl": "*",
"ext-json": "*",
"ext-libxml": "*",
"ext-mailparse": "*",
"ext-mbstring": "*",
"ext-mysqlnd": "*",
"ext-pcntl": "*",
"ext-pdo": "*",
"ext-soap": "*",
"ext-sodium": "*",
"ext-zip": "*",`

Some of these are not standard in the ddev webserver php, notably mailparse and some others. I tried adding php-mailparse to webimage_extra_packages, but this did nothing. I assumed it did not activate on compile/runtime, but I have no idea how to fix that. I tried adding a Dockerfile in the web-build folder that I copied from a colleague that uses standard docker (for a webdevops/php-apache-dev:8.1 image), but that errors on all kinds of commands.

Any suggestions on how to get those extensions working? Thanks in advance!

**Comment by rfay** [comment, marked as answer]
Not sure why you asked this twice (here and https://github.com/drud/ddev/discussions/4684) so we'll go with it here. 

The docs are in https://ddev.readthedocs.io/en/latest/users/extend/customizing-images/#adding-extra-debian-packages-with-webimage_extra_packages-and-dbimage_extra_packages

But you almost certainly need `php<phpversion>-xxx`. You'll find that most of what you need (like bcmath, curl, fileinfo, intl, json) are already in DDEV.  `ddev php -i` will show you that. 

To add `mailparse`:

`webimage_extra_packages: [php8.1-mailparse]`

Change the php version as needed.

**Reply by tomderoo** [reply]
Sorry about the double-post: internet went down at home, didn't see any response and couldn't reload the page.

Thanks for your answer. I already tried those docs, and as I mentioned I already tried adding mailparse through webimage_extra_packages. I got disheartened. But as per your answer I retried these things:

- `[php-mailparse]` for php versions 7.4 and 8.1 (just chose those as an experiment) => did not work (no errors, but also no mailparse active)
- `[php8.1-mailparse]` for php version 8.1 => did not work (error during ddev startup)
- BUT: `[php7.4-mailparse]` for php version 7.4 => works! I can see it enabled in `ddev php -i | grep mailparse`

I probably did some other thing wrong that made 8.1 error, I'm going to look into that later. Thanks for your help!

**Reply by rfay** [reply]
I know it's complicated sorting out the variety of debian packages.
